lib/action_view/component/base.rb

```diff
@@ -37,7 +37,7 @@ class Base < ActionView::Base
       # <span title="greeting">Hello, world!</span>
       #
       def render_in(view_context, *args, &block)
-        self.class.compile
+        self.class.compile!
```

Comment:
Use the version of compile that throws. This ensures the old behavior of getting exceptions for various mistakes